### PR TITLE
don't write file when applying a patch during update

### LINKED ISSUE #2014: nf-core modules update changed md5sum of skipped module in modules.json
### Description of the bug

Hey!

I ran `nf-core modules update -a` and then selected the second option (show diff and y/N before applying it). For two modules the md5sum was changed, even though I selected N and skipped the update:

see here:

https://github.com/nf-core/sarek/pull/822/files#diff-97c7fe3fe8f628d2435d00b93af4cb47052e8e1f3d33b7334d1a031fd15415d7

### Command used and terminal output

```console
nf-core modules update -a
```

### System information

Using nf-core/tools 2.6

### PR DESCRIPTION
**Pull request by mirpedrol**
Close #2014 

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Review by mashehu — APPROVED** [review]

**Comment by mashehu** [comment]
Could you maybe add a test for this?

**Comment by mirpedrol** [comment]
> Could you maybe add a test for this?

good idea :) I will add it before merging

File: tests/modules/update.py [hunk 1]
```diff
@@ -288,6 +292,67 @@ def test_update_different_branch_mix_modules_branch_test(self):
...
+    Check that the sha in modules.json is not changed."""
+    modules_json = ModulesJson(self.pipeline_dir)
+    update_obj = ModuleUpdate(self.pipeline_dir, update_all=True, show_diff=True)
+
+    assert update_obj.update() is True
+
+    mod_json = modules_json.get_modules_json()
+    # Loop through all modules and check that they are NOT updated (according to the modules.json file)
+    # Modules that can be updated but shouldn't are custom/dumpsoftwareversions and fastqc
+    # Module multiqc is already up to date so don't check
```
Comment (mashehu) [inline tests/modules/update.py:309]: Wasn't the problem that it touched files, even when we excluded them, so should we maybe also check that Multiqc didn't change?

File: tests/modules/update.py [hunk 2]
```diff
@@ -288,6 +292,67 @@ def test_update_different_branch_mix_modules_branch_test(self):
...
+
+    # Modify fastqc module, it will have a patch which will be applied during update
+    # We modify fastqc because it's one of the modules that can be updated and there's another one before it (custom/dumpsoftwareversions)
+    module_path = Path(self.pipeline_dir, "modules", "nf-core", "fastqc")
+    main_path = Path(module_path, "main.nf")
+    with open(main_path, "r") as fh:
+        lines = fh.readlines()
+    for line_index in range(len(lines)):
+        if lines[line_index] == "    label 'process_medium'\n":
+            lines[line_index] = "    label 'process_low'\n"
```
Comment (mashehu) [inline tests/modules/update.py:333]:
```suggestion
            lines[line_index] = "    label 'process_low'\n"
            break
```

**Review by mashehu — APPROVED** [review]

File: tests/modules/update.py (hunk 1 above)
Comment (mirpedrol) [reply tests/modules/update.py:309]: The problem was that it was updating the sha in `modules.json` but not updating the files. But we could also check for files unchanged to have more control

**Pull request merged**